src/main/java/net/minecraftforge/registries/ForgeRegistry.java

```diff
@@ -188,19 +189,27 @@ public ResourceLocation getKey(V value)
     @Override
     public Set<ResourceLocation> getKeys()
     {
-        return ImmutableSet.copyOf(this.names.keySet());
+        return Collections.unmodifiableSet(this.names.keySet());
     }
 
+    @Deprecated
```

Comment:
Maybe add a javadoc comment saying "Use getValuesColection() instead"

Reply:
Sure, that's a good practice.